fabric-entity-events-v1/src/main/java/net/fabricmc/fabric/api/entity/event/v1/EntityElytraEvents.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.entity.event.v1;
+
+import net.minecraft.entity.EquipmentSlot;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.item.ElytraItem;
+import net.minecraft.item.ItemStack;
+import net.minecraft.world.event.GameEvent;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * Events related to elytra flight for living entities. Elytra flight is also knows as "fall flying".
+ */
+public class EntityElytraEvents {
```

Comment:
Perhaps a `DENY` event so elytra can be conditionally disabled? Say, if I made a custom dimension and didn't want players to be able to use elytra within it?

Reply:
Ah, just noticed you yourself mentioned a possible `ALLOW` event.

Reply:
I don't think it's realistically possible atm due to the various injects that mods do. Trying to short-circuit the vanilla path is a bit tough when 3-4 mods already inject into it. Once mods migrate to this event, introducing ALLOW will be a lot easier as we will be the only ones editing vanilla code.

Reply:
Nevermind, added an `ALLOW` event. 😆